backend/query_conversion.go

```diff
@@ -0,0 +1,27 @@
+package backend
+
+import "context"
+
+// QueryConversionHandler is an EXPERIMENTAL service that allows converting queries between versions
+
+type QueryConversionHandler interface {
+	// ConvertQuery is called to covert queries between different versions
+	ConvertQuery(ctx context.Context, req *QueryConversionRequest) (*QueryConversionResponse, error)
+}
+
+type ConvertQueryFunc func(ctx context.Context, req *QueryConversionRequest) (*QueryConversionResponse, error)
+
+// QueryConversionRequest supports converting a query from on version to another
+type QueryConversionRequest struct {
+	PluginContext PluginContext `json:"pluginContext"`
+	// Queries to convert. This contains the full metadata envelope.
+	Query DataQuery `json:"query"`
+}
+
+type QueryConversionResponse struct {
+	// Converted query. It should extend v0alpha1.Query
+	Query any `json:"query"`
```

Comment:
Note: I had to use a explicit `any` here rather than make this a generic struct (`type QueryConversionResponse[T any] struct`) because either the generic conversion bubbles up until the [Manage](https://github.com/grafana/grafana-plugin-sdk-go/blob/queryConversionHandler/backend/datasource/manage.go/#L31) function and therefore becomes a breaking change, or the type is specifically set to `any` in any of the intermediate structs but then the plugin needs to explicitly use also `any` on its implementation, making the use of generics unnecessary.

Let me know if I am missing something here.